### PR TITLE
checker: Use a couple of Rust idioms

### PR DESCRIPTION
**Pull request by jameysharp**
This should have no functional change, just makes the source slightly easier to read and reason about.

Something I noticed while reviewing #113.

### CODE REVIEW

**Review by cfallin — APPROVED** [review]
Thanks! Will wait to merge this until after #113 just in case there's a merge conflict (adjacent lines touched).

**Review by elliottt — APPROVED** [review]
Nice!

**Pull request merged**
